agent/utils/compare_versions.go

```diff
@@ -164,8 +164,8 @@ func compareSemver(lhs, rhs semver) int {
 	return 0
 }
 
-// extractVersion extracts a matching version from the version number string
-func extractVersion(input string) string {
+// ExtractVersion extracts a matching version from the version number string
```

Comment:
This file is moved to use the `version.Matches`(previously in `agent/functional_tests/utils/`) in the `agent/engine` package.  And it cause the `ExtractVersion` moved out of the `agent/functional_tests/utils` so it should be made public to be used back in the functional test.